plugins/modules/zos_job_output.py

```diff
@@ -80,7 +80,7 @@
 RETURN = r"""
 jobs:
   description:
-      List of jobs output.
+      List of jobs output, or a single NOTFOUND record.
```

Comment:
I don't believe this description is accurately telling our users the outcome.

If a job is not found (assuming by name or id - I have to trace to code to see if its both) that the result would return a return code 00 and a msg "Job Not Found" or "No jobs found" , we don't return a NOTFOUND anymore for a job but it seems you still have it code for a job stepname which I commented below maybe we can just use "None" for a job step name when the job is not found. Either way I think we have some work to do here in the description/explanation. 

Couple of other things, maybe was you write the update you will see maybe why I was hesitant on supporting the plural an singular use case because now we need to instruct users with two different outcomes of a message depending on if its plural or not (depends which module is executing the code, i understand the plural use case) but lets talk about this. 

To be honest I am not even sure we should return a 00, but maybe because the module did not fail we should) for job["ret_code"]["msg_code"] = "00"